packages/react/src/reconciler.ts

```diff
@@ -73,23 +92,39 @@ const reconciler = reactReconciler<
         continue;
       }
 
+      const oldValue = oldProps[key];
+      const newValue = newProps[key];
+
+      // oldProps is the actual props that were passed in, not the adjustments
+      // we made to those props in createInstance() or previous commitUpdate()s.
+      // When we have an old prop that is a function, we therefore need to get
+      // the actual value of that prop from the remote-ui instance, which will be
+      // the proxied version of the function (see `wrapFunctionToHotSwapImplementation()`
+      // for details on why it is necessary to hot-swap functions).
+      if (typeof oldValue === 'function') {
+        const oldValueOnRemoteProps = (instance.props as any)[key];
+
+        if (
+          oldValueOnRemoteProps?.[FUNCTION_CURRENT_IMPLEMENTATION_KEY] ===
+          oldValue
+        ) {
+          needsUpdate = true;
+          hotSwapFunctions.push([
```

Comment:
Do we have to clean this array up at any point? Also, if the extension is passing in a new function every time would we have a performance issue when looping through the array? Ex.
 ```
<Button onPress={() =>{}} />
```

Reply:
No, these are just collected, then used in `commitUpdate` below this line, at which point they are garbage collected

Reply:
Oh i see it now, the array is created only when preparing to make update